### PR TITLE
feat(remodel): call ubergen to create aws-cdk-lib

### PR DESCRIPTION
**Pull request by MrArnoldPalmer**
Updates the remodel package to correctly generate the `aws-cdk-lib` package source code. Exposes a bunch of functions from within the `ubergen` package, which `remodel` now calls in order to copy all of the expected packages from the `packages/@aws-cdk` directory into `packages/aws-cdk-lib/lib` while rewriting import paths within the source code.

Additionally, `ubergen` returns a list of the bundled packages, which we use to build a map of the output directories to the correct cloudformation resource scopes. This is necessary because some scopes are repeated throughout different packages and are generated different depending on whether they are generated within a package that has other potentially related scopes, or not. This scope map is output to `aws-cdk-lib/scripts/scope-map.json` for use during code generation.

`cfn2ts` now reads this scope map, and will update it with newly generated directories/scopes as needed.

The `verify-imports-resolve-same` script currently fails. Investigating this separately as I believe the outputs are correct and it may just be a matter of adding additional entries to package.json `exports` field.

### CODE REVIEW

File: tools/@aws-cdk/remodel/lib/cli.ts [hunk 1]
```diff
@@ -77,45 +83,199 @@ async function copyTemplateFiles(src: string, target: string) {
...
+
+  // await cleanupPackages(packagesToBundle);
+  // await cleanupPackages(deprecatedPackages);
+}
+
+// async function cleanupPackages(libraries: readonly LibraryReference[]) {
+//   for (const library of libraries) {
+//     await fs.remove(library.root);
+//   }
+// }
```
Comment (TheRealAmazonKendra) [inline tools/@aws-cdk/remodel/lib/cli.ts:208]: Comment cleanup needed?

**Review by TheRealAmazonKendra — COMMENTED** [review]
I don't think(?) we can integ test this so adding an exemption. Probably should have some unit tests, though.

File: tools/@aws-cdk/remodel/lib/cli.ts (hunk 1 above)
Comment (MrArnoldPalmer) [reply tools/@aws-cdk/remodel/lib/cli.ts:208]: yeah still WIP here. Will pull out of draft when ready for review.

File: tools/@aws-cdk/remodel/lib/cli.ts [hunk 2]
```diff
@@ -77,32 +90,121 @@ async function copyTemplateFiles(src: string, target: string) {
...
+  for (const library of packagesToBundle) {
+    const libDir = path.join(libRoot, library.shortName);
+
+    // May already exist as a result of codegen
+    await createIfNoDir(libDir);
+    await transformPackage(library, pkgJson, libDir, packagesToBundle, ubgConfig.monoPackageRoot);
+    copyToExports(library, indexStatements, lazyExports, pkgJsonExports);
+  }
+
+  // make the exports.ts file pass linting
```
Comment (madeline-k) [inline tools/@aws-cdk/remodel/lib/cli.ts:147]: do you mean `index.ts`?

File: tools/@aws-cdk/remodel/lib/cli.ts [hunk 3]
```diff
@@ -77,32 +90,121 @@ async function copyTemplateFiles(src: string, target: string) {
...
+  lazyExports.unshift('/* eslint-disable @typescript-eslint/no-require-imports */');
+
+  await fs.writeFile(path.join(libRoot, 'index.ts'), indexStatements.join('\n'), { encoding: 'utf8' });
+  await fs.writeFile(path.join(libRoot, 'lazy-index.ts'), lazyExports.join('\n'), { encoding: 'utf8' });
+
+  // Filter out packages we are bundling from dev dependencies
+  const packagesToBundleName = packagesToBundle.map(p => p.packageJson.name);
```
Comment (madeline-k) [inline tools/@aws-cdk/remodel/lib/cli.ts:154]:
```suggestion
  const packagesToBundleNames = packagesToBundle.map(p => p.packageJson.name);
```

File: tools/@aws-cdk/ubergen/lib/index.ts [hunk 4]
```diff
@@ -7,20 +7,21 @@ import * as awsCdkMigration from 'aws-cdk-migration';
...
+}
 
-export async function main() {
-  console.log(`🌴  workspace root path is: ${ROOT_PATH}`);
-  const uberPackageJson = await fs.readJson(UBER_PACKAGE_JSON_PATH) as PackageJson;
-  const libraries = await findLibrariesToPackage(uberPackageJson);
-  await verifyDependencies(uberPackageJson, libraries);
-  await prepareSourceFiles(libraries, uberPackageJson);
-  await combineRosettaFixtures(libraries, uberPackageJson);
+export async function main(config: Config) {
```
Comment (madeline-k) [inline tools/@aws-cdk/ubergen/lib/index.ts:18]: Do we need these changes to main? I don't see them used in the remodel cli.ts, but maybe I am missing something.

File: tools/@aws-cdk/remodel/lib/cli.ts [hunk 5]
```diff
@@ -77,32 +90,121 @@ async function copyTemplateFiles(src: string, target: string) {
...
+    await createIfNoDir(libDir);
+    await transformPackage(library, pkgJson, libDir, packagesToBundle, ubgConfig.monoPackageRoot);
+    copyToExports(library, indexStatements, lazyExports, pkgJsonExports);
+  }
+
+  // make the exports.ts file pass linting
+  lazyExports.unshift('/* eslint-disable @typescript-eslint/no-require-imports */');
+
+  await fs.writeFile(path.join(libRoot, 'index.ts'), indexStatements.join('\n'), { encoding: 'utf8' });
+  await fs.writeFile(path.join(libRoot, 'lazy-index.ts'), lazyExports.join('\n'), { encoding: 'utf8' });
```
Comment (madeline-k) [inline tools/@aws-cdk/remodel/lib/cli.ts:151]: Can you explain this?

**Pull request closed without merging**